### ISSUE TITLE
No more Settings -> Deploy Keys on an unitialized repository

### ISSUE THREAD

**Issue by hweidner** [go-gitea/gitea#30620]
### Description

When creating a new uninitialized repository, the _Settings_ menu does not offer the _Deploy Keys_ option any more. Only _Repository_, _Collaborators_ and _Webhooks_ are shown. Older repositories also offer _Branches_, _Tags_, _Deploy Keys_, and _LFS_.

The behaviour has changed within the last versions. When I created a new uninitialized repository some weeks ago, it was possible to add a _Deploy Key_ via the _Settings_ menu.

I cannot tell if the bug is reproducible on try.gitea.io because it times out when I try to register there.

### Gitea Version

1.21.11

### Can you reproduce the bug on the Gitea demo site?

No

### Screenshots

The Settings menu on a newly created repository.

![grafik](https://github.com/go-gitea/gitea/assets/5664951/5730edfb-2c11-4b04-a8ae-381f8c5aed42)

### Operating System

Debian 12 with Docker 20.10.24+dfsg1

### How are you running Gitea?

I am using the official Docker image gitea/gitea:1 with automatic updates via watchtower.

The commandline is
```
# docker run -d --restart=always --name gitea -p 22:22 -p 3000:3000 -v /srv/gitea/data:/data gitea/itea:1
```

### Database

MySQL/MariaDB

**Comment by lunny** [comment]
I think this change is by design. The logic is if there is no branches, so we don't need UI to do that. From my current situation, maybe protected branches/tags are necessary before there is any code.

**Comment by hweidner** [comment]
> The logic is if there is no branches, so we don't need UI to do that.

I understand your argument regarding _Branches_ and _Tags_. But _Deploy Keys_ should be still configurable even to an empty (uninitialized) repository. To create an uninitialized reposory in Gitea, enter an SSH key and then push an existing repository over SSH is a perfectly valid workflow.

**Comment by lunny** [comment]
Closed by #30635
